src/store/singing.ts

```diff
@@ -508,6 +516,61 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
     },
   },
 
+  SET_PITCH_EDIT_DATA: {
+    mutation(
+      state,
+      { data, startFrame }: { data: number[]; startFrame: number },
+    ) {
+      const pitchEdit = state.tracks[selectedTrackIndex].pitchEdit;
+      let tempData = [...pitchEdit.data];
+      const endFrame = startFrame + data.length;
+      if (tempData.length < endFrame) {
+        const zeros = new Array(endFrame - tempData.length).fill(0);
+        tempData = tempData.concat(zeros);
+      }
+      for (let i = 0; i < data.length; i++) {
+        tempData[startFrame + i] = data[i];
+      }
+      state.tracks[selectedTrackIndex].pitchEdit = {
+        data: tempData,
+        frameRate: pitchEdit.frameRate,
+      };
+    },
+  },
+
+  ERASE_PITCH_EDIT_DATA: {
+    mutation(
+      state,
+      { startFrame, frameLength }: { startFrame: number; frameLength: number },
+    ) {
+      const pitchEdit = state.tracks[selectedTrackIndex].pitchEdit;
+      const tempData = [...pitchEdit.data];
+      const endFrame = Math.min(startFrame + frameLength, tempData.length);
+      for (let i = startFrame; i < endFrame; i++) {
+        tempData[i] = 0;
+      }
+      state.tracks[selectedTrackIndex].pitchEdit = {
+        data: tempData,
+        frameRate: pitchEdit.frameRate,
+      };
```

Comment:
こちらももしかしたらこれだけで良いかも･･･？
```suggestion
    const pitchEdit = state.tracks[selectedTrackIndex].pitchEdit;
    const endFrame = Math.min(startFrame + frameLength, pitchEdit.data.length);
    pitchEdit.data.fill(0, startFrame, endFrame);
```

Reply:
こちらも一旦そのままで…！